tests/src/tools/ReadyToRun.SuperIlc/Application.cs

```diff
@@ -16,25 +16,25 @@ public class Application
 
         private List<string> _mainExecutables;
 
+        private List<string> _executionScripts;
```

Comment:
When I squint at this file I can almost see factoring out the whole execution script instead of executable launch as a specialization of this class. Ideally the cmd details would be contained in a single super-class to keep complexity of this tool in check. I'm not going to hold up check-in since this is a test tool and not product code.

Reply:
Chatted offline. With today functionality making Application itself abstract is complicated because Application can run an arbitrary set of exe and cmd executions. Perhaps the biggest problem is that, after the execution generalizations, the class name "Application" has become a bit of a misnomer as now it rather means "a folder containing given sets of assemblies to compile and execute".

For now I'm proposing renaming the Application class to BuildFolder and the ApplicationSet class to BuildFolderSet, that will make the intent much more obvious. We'll tackle additional abstractions regarding execution of scripts vs. executables in a follow-up change as the exact design for this is not yet completely clear.